File: backEnd/restaurant/src/main/java/com/restaurant/model/form/Item.java
package com.restaurant.model.form;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "items")
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "image")
    private String image;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "price")
    private BigDecimal price;


    @ManyToOne
    @JoinColumn(name = "request_order_id")
    private Request requestOrder;

}

File: frontEnd/restaurant-app/src/app/model/form/item.ts
import { CartOrder } from "../cart-order";

export class Item {
  image?: string;
  quantity?: number;
  price?: number;

  constructor(cartOrder:CartOrder){
    this.image = cartOrder.image;
    this.quantity = cartOrder.quantity;
    this.price = cartOrder.price;
  }

}
